pkg/asset/manifests/aws/zones.go

```diff
@@ -0,0 +1,269 @@
+}
+
+// setZonesManagedVPC creates the CAPI NetworkSpec.Subnets setting the
+// desired subnets from install-config.yaml in the BYO VPC deployment.
+func setZonesBYOVPC(in *zoneConfigInput) error {
+	privateSubnets, err := in.Meta.PrivateSubnets(context.TODO())
+	if err != nil {
+		return fmt.Errorf("failed to get private subnets: %w", err)
+	}
+	for _, subnet := range privateSubnets {
+		in.Cluster.Spec.NetworkSpec.Subnets = append(in.Cluster.Spec.NetworkSpec.Subnets, capa.SubnetSpec{
+			ID:               subnet.ID,
+			CidrBlock:        subnet.CIDR,
+			AvailabilityZone: subnet.Zone.Name,
+			IsPublic:         subnet.Public,
+		})
+	}
+
+	publicSubnets, err := in.Meta.PublicSubnets(context.TODO())
+	if err != nil {
+		return fmt.Errorf("failed to get public subnets: %w", err)
+	}
+	for _, subnet := range publicSubnets {
+		in.Cluster.Spec.NetworkSpec.Subnets = append(in.Cluster.Spec.NetworkSpec.Subnets, capa.SubnetSpec{
+			ID:               subnet.ID,
+			CidrBlock:        subnet.CIDR,
+			AvailabilityZone: subnet.Zone.Name,
+			IsPublic:         subnet.Public,
+		})
+	}
+
+	edgeSubnets, err := in.Meta.EdgeSubnets(context.TODO())
+	if err != nil {
+		return fmt.Errorf("failed to get edge subnets: %w", err)
+	}
+	for _, subnet := range edgeSubnets {
+		in.Cluster.Spec.NetworkSpec.Subnets = append(in.Cluster.Spec.NetworkSpec.Subnets, capa.SubnetSpec{
+			ID:               subnet.ID,
+			CidrBlock:        subnet.CIDR,
+			AvailabilityZone: subnet.Zone.Name,
+			IsPublic:         subnet.Public,
+		})
+	}
+
+	vpc, err := in.Meta.VPC(context.TODO())
+	if err != nil {
+		return fmt.Errorf("failed to get VPC: %w", err)
+	}
+	in.Cluster.Spec.NetworkSpec.VPC = capa.VPCSpec{
+		ID: vpc,
+	}
+
+	return nil
+}
+
+// setZonesManagedVPC creates the CAPI NetworkSpec.VPC setting the
+// desired zones from install-config.yaml in the managed VPC deployment.
+func setZonesManagedVPC(in *zoneConfigInput) error {
+
+	out, err := extractZonesFromInstallConfig(in)
+	if err != nil {
+		return fmt.Errorf("failed to get availability zones: %w", err)
+	}
+
+	mainCIDR := capiutils.CIDRFromInstallConfig(in.InstallConfig)
+	in.Cluster.Spec.NetworkSpec.VPC = capa.VPCSpec{
+		CidrBlock: mainCIDR.String(),
+	}
+
+	// Base subnets considering only private zones, leaving one free block to allow
+	// future subnet expansions in Day-2.
+	numSubnets := len(out.AvailabilityZones) + 1
+
+	// Public subnets consumes one range from base blocks.
+	isPublishingExternal := in.Config.Publish == types.ExternalPublishingStrategy
+	publicCidrIndex := len(out.AvailabilityZones)
+	if isPublishingExternal {
+		numSubnets++
+	}
+
+	privateCIDRs, err := utilscidr.SplitIntoSubnetsIPv4(mainCIDR.String(), numSubnets)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve CIDR blocks for all private subnets: %w", err)
+	}
+	var publicCIDRs []*net.IPNet
+	if isPublishingExternal {
+		publicCIDRs, err = utilscidr.SplitIntoSubnetsIPv4(privateCIDRs[publicCidrIndex].String(), publicCidrIndex)
```

Comment:
```suggestion
		publicCIDRs, err = utilscidr.SplitIntoSubnetsIPv4(privateCIDRs[publicCidrIndex].String(), len(out.AvailabilityZones))
```
I know they are the same value, but this is easier to read for me.